stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/AddChannelFragment.kt

```diff
@@ -67,51 +67,29 @@ class AddChannelFragment : Fragment() {
 
     private fun bindAddChannelView() {
         addChannelViewModel.apply {
+            bindView(binding.addChannelView, viewLifecycleOwner)
             state.observe(viewLifecycleOwner) { state ->
+                // Handle unique states
                 when (state) {
-                    AddChannelViewModel.State.Loading -> {
-                        binding.addChannelView.showLoadingView()
-                        binding.addChannelView.hideUsersRecyclerView()
-                        binding.addChannelView.hideEmptyStateView()
-                    }
-                    AddChannelViewModel.State.Empty -> {
-                        binding.addChannelView.hideUsersRecyclerView()
-                        binding.addChannelView.hideLoadingView()
-                        binding.addChannelView.showEmptyStateView()
-                    }
-                    is AddChannelViewModel.State.Result -> {
-                        binding.addChannelView.setUsers(state.users)
-                        binding.addChannelView.hideLoadingView()
-                        binding.addChannelView.hideEmptyStateView()
-                        binding.addChannelView.showUsersRecyclerView()
-                    }
-                    is AddChannelViewModel.State.ResultMoreUsers -> {
-                        binding.addChannelView.addMoreUsers(state.users)
-                    }
                     is AddChannelViewModel.State.ShowChannel -> initializeChannel(state.cid)
                     AddChannelViewModel.State.HideChannel -> cleanChannel()
                     is AddChannelViewModel.State.NavigateToChannel -> findNavController().navigateSafely(
                         AddChannelFragmentDirections.actionOpenChat(state.cid, null)
                     )
+                    AddChannelViewModel.State.Loading,
+                    AddChannelViewModel.State.Empty,
+                    is AddChannelViewModel.State.Result,
+                    is AddChannelViewModel.State.ResultMoreUsers -> Unit
```

Comment:
Did these become unused now, should we remove them?

Reply:
The VM is reused between to *AddChannelFrahments. Some states are common for both of them (and they are handled in bind view function) and some of them are not (handled directly in AddChsnnelFragment). That's why we have Unit here. View model design could be done in a better way, but I would leave it for now as this is a part of sample app and we can come back to this later on